use-cases/virgo/synthetic_data_gen/file_gen.py

```diff
@@ -1,24 +1,53 @@
 import argparse
+import sys
 from pathlib import Path
 
 import numpy as np
 import pandas as pd
+import h5py
 from gwpy.timeseries import TimeSeries
+from time import time
+from tqdm import tqdm
 
-from ..src.dataset import generate_cut_image_dataset
+# sys.path.append(str(Path("..").resolve()))
+sys.path.append(str(Path.cwd().resolve()))
+
+from src.dataset import generate_cut_image_dataset
+
+
+def append_to_hdf5_dataset(
+    file_path: Path,
+    dataset_name: str,
+    array: np.ndarray,
+    expected_datapoint_shape: tuple,
+):
+    if tuple(array.shape[1:]) != expected_datapoint_shape:
+        actual_shape_str = ", ".join(str(s) for s in array.shape)
+        expected_shape_str = ", ".join(str(s) for s in expected_datapoint_shape)
+        raise ValueError(
+            f"'array' has an incorrect shape: ({actual_shape_str}). "
+            f"Should have been (x, {expected_shape_str})."
+        )
+
+    print(f"Appending to file: '{str(file_path.resolve())}'.")
+    with h5py.File(file_path, "a") as f:
+        dset = f[dataset_name]
+        dset.resize(dset.shape[0] + array.shape[0], axis=0)
+        dset[-array.shape[0] :] = array
```

Comment:
is this thread safe? Perhaps HDF5 is ensuring it somehow?

Reply:
I doubt it's thread safe, but I am using a single file per process so I don't think it should be a problem.